cdap-watchdog/src/main/java/co/cask/cdap/logging/plugins/RollingLocationLogAppender.java

```diff
@@ -0,0 +1,274 @@
+  private RollingPolicy rollingPolicy;
+
+  private String basePath;
+  private String filePath;
+  private String filePermissions;
+  private LocationManager locationManager;
+
+  public RollingLocationLogAppender() {
+    setName(getClass().getName());
+  }
+
+  @Override
+  public void start() {
+    if (validateAppenderProperties()) {
+      return;
+    }
+
+    // TODO remove if condition after removing Guice injection
+    if (locationFactory != null) {
+      try {
+        this.locationManager = new LocationManager(locationFactory, basePath, filePermissions);
+      } catch (IOException e) {
+        // not handling it since we will remove this if condition after log framework is in place
+      }
+      started = true;
+    }
+  }
+
+  @Override
+  public void doAppend(ILoggingEvent eventObject) throws LogbackException {
+    try {
+      // TODO Remove 'if' after removing Guice injection
+      if (locationFactory != null && this.locationManager == null) {
+        this.locationManager = new LocationManager(locationFactory, basePath, filePermissions);
+        started = true;
+      }
+
+      // logic from AppenderBase
+      if (!this.started) {
+        addStatus(new WarnStatus("Attempted to append to non started appender [" + name + "].", this));
+        return;
+      }
+
+      // logic from AppenderBase
+      if (getFilterChainDecision(eventObject) == FilterReply.DENY) {
+        return;
+      }
+
+      String namespaceId = eventObject.getMDCPropertyMap().get(LocationManager.TAG_NAMESPACE_ID);
+      if (!namespaceId.equals(NamespaceId.SYSTEM.getEntityName())) {
+        LocationIdentifier logLocationIdentifier = locationManager.getLocationIdentifier(eventObject
+                                                                                           .getMDCPropertyMap());
+        rollover(logLocationIdentifier, eventObject);
+        OutputStream locationOutputStream = locationManager.getLocationOutputStream(logLocationIdentifier, filePath);
+        setOutputStream(locationOutputStream);
+        writeOut(eventObject);
+      }
+    } catch (IllegalArgumentException iae) {
+      // this shouldn't happen
+      LOG.error("Unrecognized context ", iae);
+    } catch (IOException ioe) {
+      throw new LogbackException("Exception during append", ioe);
+    }
+  }
+
+  private void rollover(final LocationIdentifier logLocationIdentifier, ILoggingEvent eventObject) {
+    if (!locationManager.getActiveLocations().containsKey(logLocationIdentifier)) {
+      return;
+    }
+
+    final LocationOutputStream locationOutputStream = locationManager.getActiveLocations().get(logLocationIdentifier);
+    final Location logFileLocation = locationOutputStream.getLocation();
+    if (triggeringPolicy instanceof LocationTriggeringPolicy) {
+      ((LocationTriggeringPolicy) triggeringPolicy).setLocation(logFileLocation);
+      if (((LocationTriggeringPolicy) triggeringPolicy).isTriggeringEvent(eventObject)) {
```

Comment:
What if you cast it once and store it in a new local variable instead of keep casting?